compiler/rustc_codegen_ssa/src/back/linker.rs

```diff
@@ -816,7 +816,9 @@ impl<'a> Linker for GccLinker<'a> {
                 writeln!(f, "EXPORTS")?;
                 for symbol in symbols {
                     debug!("  _{symbol}");
-                    writeln!(f, "  {symbol}")?;
+                    // Quote the name in case it's reserved by linker in some way
+                    // (this accounts for names with dots in particular).
+                    writeln!(f, "  \"{symbol}\"")?;
```

Comment:
Is escaping of backslashes supported and if so can you escape both backslashes and the escape character itself?

Reply:
It's `r#"  "{symbol}""#`, there are no backslashes in the def file.

I'm not sure what exactly escaping is supported in def files, the only piece of documentation that I found is https://learn.microsoft.com/en-us/cpp/build/reference/rules-for-module-definition-statements?view=msvc-170 that says to enclose the problematic statements into double quotes, and that seems to work.

Reply:
> Is escaping of backslashes supported [...]

I meant quotes, not backslashes.

---

> It's r#"  "{symbol}""#, there are no backslashes in the def file.

I was thinking of the case where `symbol` contains a quote. It would be nice if we could do the correct thing for that, but if the linker doesn't support it, that is fine too. It normally doesn't happen anyway.

Reply:
Tested locally just to be sure, symbol with a double quote inside it `double"quote` doesn't work even if is surrounded by double quotes in def file `"double"quote"`.

Reply:
And `"double\"quote"` doesn't work either?

Reply:
`double\"quote` and `"double\"quote"` also do not work, the backslash doesn't seem to have any special effect in the def file format.